src/net.cpp

```diff
@@ -392,10 +390,6 @@ CNode* CConnman::ConnectNode(CAddress addrConnect, const char *pszDest, bool fCo
         pnode->nTimeConnected = GetTime();
         pnode->AddRef();
         GetNodeSignals().InitializeNode(pnode, *this);
-        {
-            LOCK(cs_vNodes);
-            vNodes.push_back(pnode);
-        }
```

Comment:
This isn't an issue for 0.14 because it's always been this way, but I really don't like that we pass this out through our signals before adding it locally.

So, no objection to this change, but I'd like to clean it up some post-0.14.